plugins/inventory/foreman.py

```diff
@@ -238,10 +296,203 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
+                for part in group_name.split('/'):
+                    group_label_parts.append(part.lower().replace(" ", ""))
+                    gname = to_safe_group_name('%s%s' % (self.get_option('group_prefix'), '/'.join(group_label_parts)))
+                    result_gname = self.inventory.add_group(gname)
+                    if parent_name:
+                        self.inventory.add_child(parent_name, result_gname)
+                    parent_name = result_gname
+                self.inventory.add_child(result_gname, host_name)
+
+            # Create ansible groups for environment, location and organization
+            for group in ['environment', 'location', 'organization']:
+                val = host.get('%s' % group)
+                if val:
+                    safe_key = to_safe_group_name('%s%s_%s' % (
+                        to_text(self.group_prefix),
+                        group,
+                        to_text(val).lower()
+                    ))
+                    env_lo_org = self.inventory.add_group(safe_key)
+                    self.inventory.add_child(env_lo_org, host_name)
+
+            for group in ['lifecycle_environment', 'content_view']:
+                val = content_facet_attributes.get('%s_name' % group)
+                if val:
+                    safe_key = to_safe_group_name('%s%s_%s' % (
+                        to_text(self.group_prefix),
+                        group,
+                        to_text(val).lower()
+                    ))
+                    le_cv_group = self.inventory.add_group(safe_key)
+                    self.inventory.add_child(le_cv_group, host_name)
+            params = host_params
+
+            if self.want_hostcollections:
+                hostcollections = host.get('host_collections')
+
+                if hostcollections:
+                    # Create Ansible groups for host collections
+                    for hostcollection in hostcollections:
+                        try:
+                            host_collection_group_name = to_safe_group_name('%shostcollection_%s' % (
+                                to_text(self.group_prefix),
+                                to_text(hostcollection).lower()
+                            ))
+                            hostcollection_group = self.inventory.add_group(host_collection_group_name)
+                            self.inventory.add_child(hostcollection_group, host_name)
+                        except ValueError as e:
+                            self.display.warning("Could not create groups for host collections for %s, skipping: %s" % (host_name, to_text(e)))
 
-        for host in self._get_hosts():
+            # set host vars from params
+                if self.get_option('want_params'):
+                    filtered_params = {}
+                    for p in params:
+                        if 'name' in p and 'value' in p:
+                            filtered_params[p['name']] = p['value']
```

Comment:
this doesn't work, as the `params` we get from the Reports API don't have name/value in there, but are just a dict: `{'testparam1': 'testvalue1', 'testparam2': 'testvalue2', 'remote_execution_ssh_user': 'root', 'remote_execution_effective_user_method': 'sudo', 'remote_execution_connect_by_ip': False}`

Reply:
the fix is to use `params` directly in the later code, not doing the `filtered_params` thing